roles/jboss_ws/tasks/main.yml

```diff
@@ -0,0 +1,128 @@
+---
+
+- name: internal_host_started_processing_role
+  set_fact:
+    internal_host_started_processing_role: "jboss_ws"
+
+# ----------------------------- FIND JWS_HOME ---------------------------------
+
+- name: search for folder above TOMCAT_HOME listed in tomcat process
+  raw: ps -ef | grep tomcat | grep java |awk -F"catalina.base=" '{split($2,a," ");print FS a[1]}' |  awk 'BEGIN {FS="/";OFS="/";}{ $1=""; $NF="";print}'
+  register: internal_jws_find_home_from_tomcat_process
+  ignore_errors: yes
+  when: 'jboss_ws'
+
+# Only works if locate is installed
+- name: search for folder above tomcat home with locate
+  raw: locate tomcat?/bin/startup.sh | awk -F 'tomcat' '{print $1}' | head -1
+  register: internal_jws_find_home_from_tomcat_file
+  ignore_errors: yes
+  when: 'user_has_sudo and internal_have_locate and jboss_ws'
+
+# If client has not renamed default JWS_HOME folder and followed recommended zip file install instructions, JWS_HOME should be in /opt.
+- name: search for JWS_HOME in search directories
+  raw: find {{search_directories}} -type d 2>/dev/null | egrep '*(ews|jws)*[0-99]\.[0-99]$' 
+  register: internal_jws_find_home_from_search
+  ignore_errors: yes
+  when: 'jboss_ws_ext'
+
+# Find directory above tomcat home with find
+
+- name: combine JWS_HOME candidates into a list
+  set_fact:
+    jws_home_candidates: "{{
+        (internal_jws_find_home_from_search.get('stdout_lines', []) +
+         internal_jws_find_home_from_tomcat_file.get('stdout_lines', []) +
+         internal_jws_find_home_from_tomcat_process.get('stdout_lines', []))|difference(['']) | unique }}"
+  ignore_errors: yes
+  when: 'jboss_ws'
```

Comment:
Doesn't this require JBoss Web Server to be running?  Can't we detect it using extended search when it is not running?

Reply:
Yeah sorry, added the result from extended for home and forgot to push the commit. I'll add a check for the JWS_HOME environment variable too, in case they have it set.